### ISSUE TITLE
Task1 solution sharING

### ISSUE THREAD

**Issue by SoluMilken** [discoverychallenge/dc-evaluator-2016#20]
Hi everyone, 
I am one of participants. 
In task1, we combined the transaction and user dataset into a huge matrix (number of bank * number of  user in 2014, number of features) as our training dataset. The features we extracted are 
(1) user gender, 
(2) user age category, 
(3) user income category, 
(4) user wealth category in 1~6 month,
(5) whether user has credict card in 1~6 month,
(6) user location geo information x and y,
(7) user location category,
(8) number of pos visits in first half of 2014 per user,
(9) number of webshop visits in first half of 2014 per user,
(10) number of amount of spent low, mid, high in first half of 2014 per user,
(11) number of transaction in market category a to j in first half of 2014 per user,
(12) number of transaction in the morning, afternoon, evening in first half of 2014 per user,
(13) number of credit card use in first half of 2014 per user,
(14) number of debit card use in first half of 2014 per user,
(15) the distance between user location and branch location,
(16) the distance between user social center location and branch location 
        (the social center of each user is determined by 
          sum( number of transaction in "a" pos * ("a" pos geo x, "a" pos geo y) ),
(17) number of branch visit decided by neighbors (based on user location geo)

**Comment by SoluMilken** [comment]
The models we chose for training are some simple regression models such as ridge regression and random forest. After putting lots of effort on tuning, we still can not get a better result. Can someone give us some suggestion on this kinds of tasks(huge, multi-label, super imbalance) ?   We appreciate for your sharing.

**Comment by tamasveress** [comment]
Hi SoluMilken,

Thanks for sharing your approach
I did not find that user attributes are very important in Task1 thus only included few: location type and age after one hot encoding and gender. 

Have you run feature importance of your random forest model? I wonder how the above features performed?

For me the best features were distance related: distance from user location, rank of this distance, distance from the "median pos" of the user. I took the median of geo x/y of all pos the user made transaction to calculate the latter. I would be curious to compare this with your social center feature.
I found that one can steal few good insights from a paper of one of the conference speakers, Ravi: http://theory.stanford.edu/~sergei/papers/wsdm15-geochoice.pdf
I expressed the "neighbours' popular branch" by taking the percentage of local population visited the certain branch. Local means the same geo x/y where I rounded the coordinates to kilometre accuracy. Unfortunately had no time to revisit this parameter.
I also introduced global popularity with taking the mean error of the initial model for each branch.

The algorithm I used was XGBoost logistic regression (binary:logistic).
I restricted the branch and user space around 60M to 1.2M by taking only the 5 closest (to either median pos or user location) as I expected that my 4gb ram will find it difficult to deal with such amount of data.

How you managed to run and tune the random forest for the whole user/bank space? I think probably with little more than 4gb ram?
